File: diglot_plugin/ios/Classes/DiglotPlugin.m
#import "DiglotPlugin.h"
#import <diglot_plugin/diglot_plugin-Swift.h>

@implementation DiglotPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftDiglotPlugin registerWithRegistrar:registrar];
}
@end

File: diglot_plugin/macos/Classes/DiglotPlugin.swift
import Cocoa
import FlutterMacOS

public class DiglotPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "diglot_plugin", binaryMessenger: registrar.messenger)
    let instance = DiglotPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
